Tests/webcomponents/js/ez-universaldiscovery.js

```diff
@@ -0,0 +1,339 @@
+                universalDiscovery.visibleMethod
+            );
+        });
+
+        it('should have a `multiple` property', function () {
+            assert.equal(
+                !!universalDiscovery.getAttribute('multiple'),
+                universalDiscovery.multiple
+            );
+        });
+
+        it('`multiple` property should be false by default', function () {
+            assert.isFalse(
+                universalDiscovery.multiple
+            );
+        });
+    });
+
+    describe('YUI stack loading', function () {
+        let renderViewSpy;
+
+        function assertViewRendered(spy, Y) {
+            assert.isOk(
+                spy.callCount
+            );
+            assert.equal(
+                spy.firstCall.args[0],
+                Y.eZ.UniversalDiscoveryView,
+                'The UniversalDiscoveryView should have been rendered'
+            );
+        }
+
+        afterEach(function () {
+            delete window.eZ;
+            renderViewSpy.restore();
+        });
+
+        describe('`ez-yui-app:ready` event', function () {
+            beforeEach(function () {
+                universalDiscovery = fixture('BasicTestFixture');
+                defineGlobals();
+                renderViewSpy = sinon.spy(window.eZ.YUI.app, 'renderSideView');
+            });
+
+            it('should render the universal discovery', function () {
+                document.dispatchEvent(new CustomEvent('ez:yui-app:ready'));
+
+                assertViewRendered(renderViewSpy, window.eZ.YUI.Y);
+            });
+        });
+
+        describe('already loaded', function () {
+            beforeEach(function () {
+                defineGlobals();
+                renderViewSpy = sinon.spy(window.eZ.YUI.app, 'renderSideView');
+                universalDiscovery = fixture('BasicTestFixture');
+            });
+
+            it('should render the universal discovery right away', function () {
+                assertViewRendered(renderViewSpy, window.eZ.YUI.Y);
+            });
+        });
+    });
+
+    describe('UDW init', function () {
+        let renderViewStub;
+        const container = document.createElement('div');
+        const location = getModelMock();
+        const content = getModelMock();
+        const contentType = getModelMock();
+
+        function assertEventParam(e) {
+            assert.equal(e.detail.selection.content, modelObject);
+            assert.equal(e.detail.selection.location, modelObject);
+            assert.equal(e.detail.selection.contentType, modelObject);
+            assert.isFalse(e.bubbles);
+        }
+
+        beforeEach(function () {
+            universalDiscovery = fixture('BasicTestFixture');
+            defineGlobals();
+        });
+
+        afterEach(function () {
+            delete window.eZ;
+            renderViewStub.restore();
+        });
+
+        it('should append the YUI view container', function () {
```

Comment:
actually most of the assertions are about the parameters. You should split that into 2 tests:
* `should append the YUI view container`
* `should have received parameters from the properties`